_posts/2021-05-03-deploying-kafka-with-lets-encrypt-certificates.md

```diff
@@ -0,0 +1,485 @@
+---
+layout: post
+title:  "Deploying Kafka with Let's Encrypt certificates"
+date: 2021-05-03
+author: jakub_scholz
+---
+
+When deploying Apache Kafka on Kubernetes, Strimzi makes it easy to configure Kafka listeners with TLS encryption.
+But out of the box, these listeners will use only certificates signed by the internal self-signed certification authority generated by Strimzi.
+When clients try to connect to a listener secured with such a certificate, they will not trust it by default.
+You will need to get the public key of the certification authority which signed the server certificates and configure the client to trust it.
+This brings a lot of complications, since you need to get the public key, distribute it to all the clients, and configure them to use it.
+And you will need to update the public key when it changes, for example, because the old one expired.
+This would be much easier if the listeners can use TLS certificates signed by a certification authority which the clients trust out of the box.
```

Comment:
```suggestion
This would be much easier if the listeners can use TLS certificates signed by a certification authority which the clients already trust.
```